data_structures/src/superblock.rs

```diff
@@ -572,6 +602,51 @@ fn magic_partition<T: Clone>(v: &[T], first: usize, size: usize) -> Vec<T> {
     v_subset
 }
 
+// Take size element out of v.len() starting with element at index first plus an offset:
+// magic_partition(v, 3, 3, r), v=[0, 1, 2, 3, 4, 5], r=[1].
+// Will return elements at index 4, 0, 2.
+fn magic_partition_2<T>(v: &[T], first: usize, size: usize, rand_distribution: &[u8]) -> HashSet<T>
+where
+    T: Clone + Eq + std::hash::Hash,
+{
+    let mut hs_subset = HashSet::default();
+    if first >= v.len() {
+        return hs_subset;
+    }
```

Comment:
Why returning an empty set in this case?

Is it just to comply with the function signature without introducing a `Result` because in theory it is impossible that `f >= v.len() == true`? Or is there any good reason for it? I'm trying to understand if this is a protocol thing or is only due to specifics of the Rust implementation.

Reply:
I believe it was used to handle the case of `v.len() == 0`, but now that should have been checked before calling this function. So this is probably dead code that can be removed.

Reply:
Yes, we can modify by an unreachable!()